iceoryx_hoofs/include/iceoryx_hoofs/memory/relative_pointer.inl

```diff
@@ -27,25 +27,24 @@ namespace memory
 template <typename T>
 // NOLINTJUSTIFICATION NewType size is comparable to an integer, hence pass by value is preferred
 // NOLINTNEXTLINE(performance-unnecessary-value-param)
-inline RelativePointer<T>::RelativePointer(ptr_t ptr, segment_id_t id) noexcept
-    : m_id(id)
-    , m_offset(computeOffset(ptr))
+inline RelativePointer<T>::RelativePointer(ptr_t const ptr, const segment_id_t id) noexcept
+    : RelativePointer(getOffset(id, ptr), id)
 {
 }
 
 template <typename T>
+// AXIVION Next Construct AutosarC++19_03-A12.1.5 : This is the main c'tor which the other c'tors use
 // NOLINTJUSTIFICATION NewType size is comparable to an integer, hence pass by value is preferred
 // NOLINTNEXTLINE(performance-unnecessary-value-param)
-inline RelativePointer<T>::RelativePointer(offset_t offset, segment_id_t id) noexcept
+inline RelativePointer<T>::RelativePointer(const offset_t offset, const segment_id_t id) noexcept
```

Comment:
Yes, here the `const` can be to the left (due to the type of `offset_t`) but this will certainly be confusing compared to other functions where this is not the case.

As it really serves no purpose, that would be another argument to just not use `const` here.

Reply:
See my comment [here](https://github.com/eclipse-iceoryx/iceoryx/pull/1765#discussion_r1011455154), closing this thread.